src/reporters/json.ts

```diff
@@ -27,18 +27,119 @@ import BaseReporter from './base';
 import { formatError, getTimestamp } from '../helpers';
 import { Journey, Step } from '../dsl';
 import snakeCaseKeys from 'snakecase-keys';
+import { NetworkInfo } from '../common_types';
 
 /* eslint-disable @typescript-eslint/no-var-requires */
-const programVersion = require('../../package.json').version;
+const { version, name } = require('../../package.json');
+
+type OutputType =
+  | 'journey/register'
+  | 'journey/start'
+  | 'step/screenshot'
+  | 'step/end'
+  | 'journey/network_info'
+  | 'journey/filmstrips'
+  | 'journey/browserconsole'
+  | 'journey/end';
+
+type OutputFields = {
+  type: OutputType;
+  journey: Journey;
+  timestamp?: number;
+  url?: string;
+  step?: Partial<Step>;
+  error?: Error;
+  root_fields?: Record<string, unknown>;
+  payload?: Record<string, unknown>;
+  blob?: string;
+};
+
+function getMetadata() {
```

Comment:
@andrewvc I picked only minimal things for metadata, if you have any ideas for other fields, we can incorporate that as well.